### DISCUSSION TITLE
Prefetch issues with optional catch-all route

### DISCUSSION THREAD

**Discussion by Bottersnike** [vercel/next.js#14263] [Help] [answered]
I've got a route setup at `pages/[[...page]].js` as a fallback route (`optionalCatchAll` is enabled). Navigating to a route that isn't explicitly defined is successfully landing on this page as would be expected, both when clicking on a link, and when using browser navigation. When hovering over a link, however, next is attempting to prefetch the page. For explicitly defined routes this works fine, however for routes that would fall back to `[[...page]].js`, requests are being made to `/_next/static/....../pages/some/route/here.js`, which then 404s. While this doesn't directly impact UX, it causes the console to be spammed with a wall of 404 errors.

Is this intended behaviour, and is there a way to change it without globally disabling prefetching?

**Comment by Timer** [comment, marked as answer]
It sounds like you're using `<Link>` incorrectly. Can you please share your component?

**Reply by Bottersnike** [reply]
I've been converting a bunch of components over from a CRA project, so I have a `Link.js` file with:

```js
import NextLink from "next/link";

export const Link =({ to, children, onClick, ...props }) => {
    return <NextLink href={to || "#"}>
        <a onClick={onClick} {...props}>{children}</a>
    </NextLink >;
};
```

I created a simple page to test it (`pages/test.js`) containing both this component and a straight `next/link` component:

```js
import { Link } from "@ractf/ui-kit";
import NextLink from "next/link";

export default () => {
    return <>
        <Link to="/">
            Home
        </Link>
        <NextLink href="/">
            <a>
                Home
            </a>
        </NextLink>
    </>;
};
```

Both links had 404 errors when preloading on hover, but navigation itself worked as expected:
![](https://i.imgur.com/H1Cw7sz.gif)

**Reply by Timer** [reply]
Your links need to be in the following shape:
```js
<NextLink href="/[[...page]]" as="/">
<NextLink href="/[[...page]]" as="/other">
```

**Reply by Bottersnike** [reply]
Ah that'd do it, tysm. Is there a way I can hook the router in my custom Link component to determine if a route resolves to an explicit path or if it'll fallback to `[[...page]].js` and apply the props accordingly?

**Reply by Timer** [reply]
No, but we're going to add this behavior to Next.js automatically soon:
#8207

**Reply by Bottersnike** [reply]
Ah cool; I'll keep my eyes peeled on that RFC :)

**Reply by ivorpad** [reply]
@Timer 

Prefetch is having issues, tho.

For instance:

`href="/store/[[...slug]]"` prefetches `/store/[products/quiksilver-mens-3-2mm-syncro-series-back-zip-gbs-wetsuit]` (notice the open/close brackets)

It's reported here: https://github.com/vercel/next.js/issues/14290

**Reply by Timer** [reply]
Will be fixed on the next canary @ivorpad (`^9.4.5-canary.15`)!
